proposals/load_ml_model.json

```diff
@@ -0,0 +1,53 @@
+{
+    "id": "load_ml_model",
+    "summary": "Load a ML model",
+    "description": "Loads a machine learning model from a STAC Item.\n\nSuch a model could be trained and saved as part of a previous batch job with processes such as ``fit_random_forest()`` and ``save_ml_model()``.",
+    "categories": [
+        "machine learning",
+        "import"
+    ],
+    "experimental": true,
+    "parameters": [
+        {
+            "name": "id",
+            "description": "The STAC Item to load the machine learning model from. The STAC Item must implement the `ml-model` extension.",
+            "schema": [
+                {
+                    "title": "URL",
+                    "type": "string",
+                    "format": "uri",
+                    "subtype": "uri",
+                    "pattern": "^https?://"
+                },
+                {
+                    "title": "Batch Job ID",
+                    "description": "Loading a model by batch job ID is possible only if a single model has been saved by the job. Otherwise, you have to load a specific model from a batch job by URL.",
+                    "type": "string",
+                    "subtype": "job-id",
+                    "pattern": "^[\\w\\-\\.~]+$"
+                },
+                {
+                    "title": "User-uploaded Files",
+                    "type": "string",
+                    "subtype": "file-path",
+                    "pattern": "^[^\r\n\\:'\"]+$"
```

Comment:
a path to user uploaded file is always a relative path I guess, so shouldn't the regex specify that it can't start with `/`?

Reply:
No, the root is the user workspace. /a.txt is simply a file in the root folder of the user's workspace. That's equivalent to a.txt and ./a.txt as the cwd is also the root.

Reply:
The API already defines these constraints that it must be relative, and `.` should be avoided:

https://github.com/Open-EO/openeo-api/blob/f303d65a3291d4cd74dacc0e796803bb5d6fa03b/openapi.yaml#L3283-L3288
https://github.com/Open-EO/openeo-api/blob/f303d65a3291d4cd74dacc0e796803bb5d6fa03b/openapi.yaml#L5668-L5680

Reply:
That's a misunderstanding: In OpenAPI we define the path in the URL, in the processes we define the path from the user-POV, which are slightly different use cases. A path should not start with a / in a URL as then you could end up with `https://example.com/files//a.txt`.